proxy/src/stream.rs

```diff
@@ -118,19 +119,21 @@ impl<S: AsyncWrite + Unpin> PqStream<S> {
     }
 }
 
-pin_project! {
-    /// Wrapper for upgrading raw streams into secure streams.
-    /// NOTE: it should be possible to decompose this object as necessary.
-    #[project = StreamProj]
-    pub enum Stream<S> {
-        /// We always begin with a raw stream,
-        /// which may then be upgraded into a secure stream.
-        Raw { #[pin] raw: S },
-        /// We box [`TlsStream`] since it can be quite large.
-        Tls { #[pin] tls: Box<TlsStream<S>> },
-    }
+/// Wrapper for upgrading raw streams into secure streams.
+/// NOTE: it should be possible to decompose this object as necessary.
+pub enum Stream<S> {
+    /// We always begin with a raw stream,
+    /// which may then be upgraded into a secure stream.
+    Raw { raw: S },
+    /// We box [`TlsStream`] since it can be quite large.
+    Tls {
+        tls: Box<TlsStream<S>>,
+        certified_key: Option<Arc<CertifiedKey>>,
+    },
 }
```

Comment:
This pinning change seems like an independent change. Maybe extract that to a separate commit?

Reply:
BTW, what does "decomposing this object as necessary" mean?

Reply:
Sure, I can split the commit.

> BTW, what does "decomposing this object as necessary" mean?

Not sure, I'd have to look up the history of that comment I think